src/collections/Node.js

```diff
@@ -142,6 +148,9 @@ var mutationMethods = {
     return this.forEach(function(path, i) {
       var newNodes =
         (typeof nodes === 'function') ? nodes.call(path, path, i) : nodes;
+      if (newNodes.__IS_JSX__) {
```

Comment:
If this was changed to:

```js
let newNodes = nodes;
if (nodes && typeof nodes.toAST === 'function') {
  newNodes = nodes.toAST()
} else if (typeof nodes === 'function') {
  newNodes = nodes.call(path, path, i);
}
```

then *any* value that implements a `toAST` method could be passed as replacement, making it possible to define custom replacers.

Same below.